src/v0/destinations/ga4/transform.js

```diff
@@ -262,12 +262,22 @@ const responseBuilder = (message, { Config }) => {
   userProperties = extractCustomFields(
     message,
     userProperties,
-    ['properties.user_properties'],
+    ['properties.user_properties', 'context.traits'],
     GA4_RESERVED_USER_PROPERTY_EXCLUSION,
   );
 
   if (!isEmptyObject(userProperties)) {
-    rawPayload.user_properties = userProperties;
+    const userPropertiesKeys = Object.keys(userProperties);
+    const validatedUserProperties = {};
+
+    for (const key of userPropertiesKeys) {
+      const value = userProperties[key];
+      if (typeof value === 'string' || typeof value === 'number') {
+        validatedUserProperties[key] = { value };
+      }
+    }
+
+    rawPayload.user_properties = validatedUserProperties;
```

Comment:
Even we are also dropping silently any properties which are not `string` or `number`

Reply:
yes, if we don't drop it then GA4 will silently drop whole event

Reply:
Lets remove this validation as we discussed with AMy we will not do this now